packages/ts/src/components/treemap/index.ts

```diff
@@ -0,0 +1,121 @@
+
+  constructor (config?: TreemapConfigInterface<Datum>) {
+    super()
+    if (config) this.setConfig(config)
+    this.tiles = this.g.append('g')
+  }
+
+  _render (customDuration?: number): void {
+    const { config, datamodel: { data } } = this
+    const duration = isNumber(customDuration) ? customDuration : config.duration
+
+    if (!config.layers?.length) {
+      console.warn('Unovis | Treemap: No layers defined')
+      return
+    }
+
+    const layerAccessors = config.layers.map(layerAccessor => (i: number) => getString(data[i], layerAccessor, i))
+    const nestedData = group(data.keys(), ...layerAccessors as [(d: number) => string])
+
+    const rootNode = config.value !== undefined
+      ? hierarchy(nestedData).sum(index => typeof index === 'number' && getNumber(data[index], config.value, index))
+      : hierarchy(nestedData).count()
+
+    const treemapLayout = treemap()
+      .size([this._width, this._height])
+      .round(true)
+      .padding(this.config.padding)
+
+    const treemapData = treemapLayout(rootNode) as TreemapNode<Datum>
+
+    const descendants = treemapData.descendants()
+
+    const opacity = scaleLinear()
+      .domain([1, max(descendants, d => d.depth)])
+      .range([1, 0.2])
+
+    treemapData
+      .eachBefore((node) => {
+        if (!node.children) return
+        node.children.forEach((child, i) => {
+          const treemapChild = child as TreemapNode<Datum>
+          const color = getColor(treemapChild, config.tileColor, i, treemapChild.depth !== 1)
+          treemapChild._state = {
+            fill: color ?? (node as TreemapNode<Datum>)._state?.fill,
+            fillOpacity: color === null ? opacity(treemapChild.depth) : null,
+          }
+        })
+      })
+
+    // Render tiles
+    const visibleNodes = descendants.filter(d => d.depth > 0)
+    const tiles = this.tiles
+      .selectAll<SVGGElement, TreemapNode<Datum>>('g')
+      .data(visibleNodes, d => d._id)
+    const tilesEnter = tiles.enter().append('g')
+
+    // Background rectangles
+    tilesEnter.append('rect').classed('background', true)
+    tiles.merge(tilesEnter).select('rect.background')
+      .style('fill', '#ffffff')
+      .call(selection => smartTransition(selection, duration)
+        .attr('x', d => d.x0)
+        .attr('y', d => d.y0)
+        .attr('width', d => d.x1 - d.x0)
+        .attr('height', d => d.y1 - d.y0)
+      )
+
+    // Foreground rectangles
+    tilesEnter.append('rect').classed('foreground', true)
+    tiles.merge(tilesEnter).select('rect.foreground')
+      .style('fill', d => d._state?.fill ?? getColor(d, config.tileColor))
+      .style('fill-opacity', d => d._state?.fillOpacity ?? 1)
+      .call(selection => smartTransition(selection, duration)
+        .attr('x', d => d.x0)
+        .attr('y', d => d.y0)
+        .attr('width', d => d.x1 - d.x0)
+        .attr('height', d => d.y1 - d.y0)
+      )
+
+    // Labels
+    if (config.tileLabel) {
```

Comment:
I feel like this might need an `else` statement. What if `tileLabel` was specified initially, but then got removed from the config?

Reply:
Great point! Will investigate that case.

Reply:
I removed the `config.tileLabel` option for now, since the labels are defined by the values returned by the layers configuration. We can add it back if the need for it becomes more clear, but as of now I'm not seeing why it is required. Perhaps as an override or something?